pkg/manifestcomparators/comp_conditions_must_have_proper_ssa_tags.go

```diff
@@ -0,0 +1,103 @@
+package manifestcomparators
+
+import (
+	"fmt"
+	"strings"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+type conditionsMustHaveProperSSATags struct{}
+
+func ConditionsMustHaveProperSSATags() CRDComparator {
+	return conditionsMustHaveProperSSATags{}
+}
+
+func (conditionsMustHaveProperSSATags) Name() string {
+	return "ConditionsMustHaveProperSSATags"
+}
+
+func (conditionsMustHaveProperSSATags) WhyItMatters() string {
+	return "Conditions should follow the standard schema included in  " +
+		"https://github.com/kubernetes/apimachinery/blob/release-1.29/pkg/apis/meta/v1/types.go#L1482-L1542" +
+		"and collection of conditions should be treated as a map with a key of type. " +
+		"This is indicated in kubebuilder tags with '// +listType=map' and '// +listMapKey=type'."
+
+}
+
+func (c conditionsMustHaveProperSSATags) Validate(crd *apiextensionsv1.CustomResourceDefinition) (ComparisonResults, error) {
+	errsToReport := []string{}
+
+	for _, newVersion := range crd.Spec.Versions {
+		conditionsWithoutMapListType := []string{}
+		conditionsWithoutListMapKeysType := []string{}
+		SchemaHas(newVersion.Schema.OpenAPIV3Schema, field.NewPath("^"), field.NewPath("^"), nil,
+			func(s *apiextensionsv1.JSONSchemaProps, fldPath, simpleLocation *field.Path, _ []*apiextensionsv1.JSONSchemaProps) bool {
+				if s.Type != "array" {
+					return false
+				}
+
+				if !strings.Contains(simpleLocation.String(), ".conditions") {
```

Comment:
I think we only want to enforce on `^.status.conditions` since that's where `kubectl wait` looks.

Reply:
OK...I remembered the API https://github.com/openshift/api/blob/master/insights/v1alpha1/types_insights.go#L111 where conditions are nested elsewhere. So I thought it could be useful